corehq/apps/app_manager/helpers/validators.py

```diff
@@ -539,6 +539,17 @@ def validate_search_config(self):
                             "details": search_config.get_instance_name()
                         }
 
+            contains_ungrouped_properties = any(not prop.group_key for prop in search_config.properties)
+            contains_grouping_property = any(prop.is_group for prop in search_config.properties)
+            if contains_ungrouped_properties and contains_grouping_property:
+                for prop in search_config.properties:
+                    if not prop.group_key:
```

Comment:
I don't love my implementation so I think that's a great idea, you mean like this?
```
ungrouped_properties = [prop for prop in search_config.properties if not prop.group_key]
contains_grouping_property = any(prop.is_group for prop in search_config.properties)
if contains_grouping_property:
    for prop in ungrouped_properties:
        yield{
        }
```

Reply:
That actually looks better than what I was thinking. You could even move `ungrouped_properties` inside the conditional at that point, since it's not needed unless `contains_grouping_property` is true.
```python
contains_grouping_property = any(prop.is_group for prop in search_config.properties)
if contains_grouping_property:
    ungrouped_properties = [prop for prop in search_config.properties if not prop.group_key]
    for prop in ungrouped_properties:
        ...
```

Reply:
Thanks for this suggestion [0b0d35e](https://github.com/dimagi/commcare-hq/pull/33828/commits/0b0d35e007344b446f85656bca248ae84f544550)